File: python/src/semantic_retrieval/retrieval/vector_dbs/vector_db_document_retriever.py
from typing import List, Optional

from semantic_retrieval.data_store.vector_dbs.vector_db import VectorDB
from semantic_retrieval.document.document import Document
from semantic_retrieval.document.metadata.document_metadata_db import (
    DocumentMetadataDB,
)
from semantic_retrieval.retrieval.vector_dbs.vector_db_retriever import (
    BaseVectorDBRetriever,
)
from semantic_retrieval.utils.callbacks import CallbackManager


class VectorDBDocumentRetriever(BaseVectorDBRetriever[List[Document]]):
    def __init__(
        self,
        vector_db: VectorDB,
        metadata_db: DocumentMetadataDB,
        callback_manager: Optional[CallbackManager] = None,
    ):
        super().__init__(vector_db, metadata_db, callback_manager)

    async def process_documents(
        self, documents: List[Document]
    ) -> List[Document]:
        return documents

File: typescript/src/retrieval/vector-DBs/vectorDBDocumentRetriever.ts
import { Document } from "../../document/document";
import { RetrieverProcessDocumentsEvent } from "../../utils/callbacks";
import {
  BaseVectorDBRetriever,
  VectorDBRetrieverParams,
} from "./vectorDBRetriever";

/**
 * Base class for retrieving Documents from an underlying VectorDB
 */
export class VectorDBDocumentRetriever extends BaseVectorDBRetriever<
  Document[]
> {
  constructor(params: VectorDBRetrieverParams) {
    super(params);
  }

  protected async processDocuments(documents: Document[]): Promise<Document[]> {
    const event: RetrieverProcessDocumentsEvent = {
      name: "onRetrieverProcessDocuments",
      documents,
    };
    await this.callbackManager?.runCallbacks(event);

    return documents;
  }
}
